pkg/operator/highcpuusagesnoalertcontroller/performanceprofile_test.go

```diff
@@ -0,0 +1,168 @@
+package highcpuusagesnoalertcontroller
+
+import (
+	"context"
+	"math/rand"
+	"strconv"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/dynamic/fake"
+)
+
+func TestPerformanceProfileControlPlaneCores(t *testing.T) {
+	tests := []struct {
+		name           string
+		clusterObjects []runtime.Object
+
+		expectedCores       int
+		expectedToFindCores bool
+		expectedErr         bool
+	}{
+		{
+			name:                "no performanceProfile",
+			expectedToFindCores: false,
+			expectedErr:         false,
+		}, {
+			name:                "one performanceProfile",
+			clusterObjects:      []runtime.Object{performanceProfileWithNodeSelector("node-role.kubernetes.io/master")},
+			expectedCores:       2,
+			expectedToFindCores: true,
+			expectedErr:         false,
+		},
+		{
+			name:                "only worker performanceProfile",
+			clusterObjects:      []runtime.Object{performanceProfileWithNodeSelector("node-role.kubernetes.io/worker")},
+			expectedToFindCores: false,
+			expectedErr:         false,
+		},
+		{
+			name: "multiple performanceProfiles",
+			clusterObjects: []runtime.Object{
+				performanceProfileWithNodeSelector("node-role.kubernetes.io/master"),
+				performanceProfileWithNodeSelector("node-role.kubernetes.io/worker")},
+			expectedCores:       2,
+			expectedToFindCores: true,
+			expectedErr:         false,
+		},
+		{
+			name:                "invalid cpu set in performance profile",
+			clusterObjects:      []runtime.Object{invalidCPUSetInvalidPerformanceProfile()},
+			expectedToFindCores: false,
+			expectedErr:         true,
+		},
+		{
+			name:                "no node selector in performance profile",
+			clusterObjects:      []runtime.Object{noNodeSelectorInvalidPerformanceProfile()},
+			expectedToFindCores: false,
+		},
+		{
+			name:                "no cpu set in performance profile",
+			clusterObjects:      []runtime.Object{noCPUSetInvalidPerformanceProfile()},
+			expectedToFindCores: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			coresFound, found, err := performanceProfileControlPlaneCores(context.Background(), clientWithObjects(tt.clusterObjects...))
+			if (err != nil) != tt.expectedErr {
+				t.Errorf("performanceProfileControlPlaneCores() error = %v, expectedErr %v", err, tt.expectedErr)
+				return
+			}
+			if coresFound != tt.expectedCores {
+				t.Errorf("performanceProfileControlPlaneCores() got = %v, want %v", coresFound, tt.expectedCores)
+			}
+			if found != tt.expectedToFindCores {
+				t.Errorf("performanceProfileControlPlaneCores() got1 = %v, want %v", found, tt.expectedToFindCores)
```

Comment:
could you rename to `t.Errorf("performanceProfileControlPlaneCores() foundCores = %v, expectedToFindCores = %v", found, tt.expectedToFindCores)` ?